src/Simulation/TargetDefinitions/Decompositions/ExpFromIsing.qs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Applies the exponential of a multi-qubit Pauli operator.
+    ///
+    /// # Description
+    /// \begin{align}
+    ///     e^{i \theta [P_0 \otimes P_1 \cdots P_{N-1}]},
+    /// \end{align}
+    /// where $P_i$ is the $i$th element of `paulis`, and where
+    /// $N = $`Length(paulis)`.
+    ///
+    /// # Input
+    /// ## paulis
+    /// Array of single-qubit Pauli values indicating the tensor product
+    /// factors on each qubit.
+    /// ## theta
+    /// Angle about the given multi-qubit Pauli operator by which the
+    /// target register is to be rotated.
+    /// ## qubits
+    /// Register to apply the given rotation to.
+    @EnableTestingViaName("Test.TargetDefinitions.Exp")
+    operation Exp (paulis : Pauli[], theta : Double, qubits : Qubit[]) : Unit is Adj + Ctl {
```

Comment:
@alan-geller Here is an example of what we were talking about on the other thread: an alternate definition of `Microsoft.Quantum.Intrinsic.Exp` that is based on the Ising rotation gates. To differentiate, I named the file `ExpFromIsing.qs`. I'm definitely open to suggestions regarding naming patterns, too. I did consider trying to keep the names simple and putting related files in subfolders together (for example, `ExpFromIsing.qs` would go with `ExpFracFromIsing`, `IsingXX`, `IsingYY`, and `IsingZZ`), but that kind of approach can quickly lead to overly complex hierarchies of folders, especially when some files are shared among only slightly related definitions. It's also possible to imagine a target definition for a system that has only one of the rotations, say `IsingXX`, and then the overlap among files would again be confusing. For those reasons, I thought keeping a single folder with flat hierarchy and using filenames to disambiguate seemed the least prone to overcomplexity.